routes/images.go

```diff
@@ -20,33 +21,34 @@ type Images struct {
 	InstanceStore store.InstanceStore
 	Executor      exec.Executor
 	Authenticator auth.Authenticator
+	Logger        log.Logger
 }
 
 func (i Images) Get(w http.ResponseWriter, r *http.Request) {
 	_, err := i.Authenticator.AuthenticateRequest(r)
 	if err != nil {
-		log.Print(err.Error())
+		i.Logger.Info(err.Error())
 		RenderError(w, http.StatusUnauthorized, unauthorizedError)
 		return
 	}
 
 	id, err := strconv.Atoi(mux.Vars(r)["id"])
 	if err != nil {
-		log.Print(err.Error())
+		i.Logger.Info(err.Error())
 		RenderError(w, http.StatusNotFound, notFoundError)
 		return
 	}
 
 	image, err := i.ImageStore.Get(id)
 	if err != nil {
-		log.Print(err.Error())
+		i.Logger.Info(err.Error())
 		RenderError(w, http.StatusNotFound, notFoundError)
 		return
 	}
 
 	err = jsonapi.MarshalOnePayload(w, &image)
 	if err != nil {
-		log.Print(err.Error())
+		i.Logger.With("error", err.Error()).With("http_request", r).Error("failed to marshal payload")
```

Comment:
We're doing this a lot. I wonder if it would be nicer to have a `HTTPLogger` which explicitly takes an http request. We then avoid the issue that one of these lines has `htpp_request` or some other typo and our log filtering then doesn't work correctly.

Reply:
Similarly with `With("error", err.Error())`. We do this everywhere, it would be a lot cleaner to just have `WithError(err)`.

Reply:
Yeah I'd like to use a default middleware chain that collects the errors if we return them and aggregates the `http_request`. I was going to leave this for another PR though- do you think it's worth doing now?

Reply:
Yeah I do actually - I don't think it would be that hard and it's not clear from this PR that there's a followup intended.